modules/sitemaps/sitemaps-provider-decorator.php

```diff
@@ -94,8 +198,17 @@ public function get_max_num_pages( $object_subtype = '' ) {
 	 * @return string The composed URL for a sitemap entry.
 	 */
 	public function get_sitemap_url( $name, $page ) {
-		$url = $this->provider->get_sitemap_url( $name, $page );
-		return str_replace( $this->provider->name, $this->name, $url );
+		$parts = explode( '-', $name );
+		$lang = end( $parts );
+		$lang = $this->model->get_language( $lang ); // Validates that we got an existing language code.
+
+		if ( $lang ) {
```

Comment:
This comment made me think that a `PLL_Model::language_exists()` method could be helpful.

Reply:
That's something we could introduce it if some day we need it. In most cases however, I use PLL_Model::get_language() to validate that the language exists and retrieve the corresponding object in one call.